lib/instrumentation/transaction.js

```diff
@@ -58,7 +58,19 @@ function Transaction (agent, name, type) {
     }
   })
 
-  this.id = uuid.v4()
+  Object.defineProperty(this, 'id', {
+    get () {
+      return this.context.id.toString('hex')
```

Comment:
I made some changes to use a shared buffer. I could maybe stringify the values out of the buffer in the constructor and store them, rather than use the getters? The downside is that it'd then be storing fields even when they aren't used. Might also be a good idea to `Object.freeze(this)` in the constructor, since the object itself _should_ be basically immutable in normal circumstances. 🤔

Reply:
I ended up changing to a getter-based lazy properties thing.